### PR TITLE
[linstor] Update Linstor. Fix D8LinstorControllerTargetDown alert

### PR DESCRIPTION
**Pull request by AleksZimin**
## Description

- Update Linstor components:
  - linstor-csi 1.0.1 -> 1.2.0
  - linstor-server 1.22.1 -> 1.23.0
  - piraeus-ha-controller 1.1.3 -> 1.1.4
  - piraeus-operator 1.10.3 -> 1.10.4
- Fix D8LinstorControllerTargetDown alert by changing the pod monitor to a service monitor for the Linstor controller. Also adds a patch to piraeus-operator that adds a metrics port to the Linstor controller service and Linstor controller endpoints.

## Why do we need it, and what problem does it solve?

We are now using old versions of Linstor components.
The D8LinstorControllerTargetDown alert now fires when a linstor controller has more than one replica. This happens because of the leader election mechanism for linstor controllers. After the leader is selected, other pods of linstor controllers don't serve any metrics

This PR fixes it

## What is the expected result?

Updated Linstor components
No alerts when Linstor controller is in HA mode

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: linstor
type: fix
summary: Update Linstor. Fix D8LinstorControllerTargetDown alert
impact_level: default
```

### Main branch: parameter securedMonitoringPort is not set
```bash
kubectl -n d8-system set image deployment deckhouse deckhouse="dev-registry.deckhouse.io/sys/deckhouse-oss:main"
kubectl -n d8-system get deployment deckhouse -o wide
kubectl get deployment linstor-controller -o yaml | grep -A1 "name: K8S_AWAIT_ELECTION_SERVICE_PORTS_JSON"
kubectl -n d8-linstor get linstorcontrollers.piraeus.linbit.com -o yaml | grep secured
kubectl -n d8-linstor get svc linstor
kubectl -n d8-linstor get endpoints linstor
kubectl -n d8-monitoring get podmonitors.monitoring.coreos.com linstor-controller
kubectl -n d8-monitoring get servicemonitors.monitoring.coreos.com linstor-controller
```
main-kubectl
main-alerts

### PR 4823: parameter securedMonitoringPort is set

```bash
kubectl -n d8-system set image deployment deckhouse deckhouse="dev-registry.deckhouse.io/sys/deckhouse-oss:pr4823"
kubectl -n d8-system get deployment deckhouse -o wide
kubectl get deployment linstor-controller -o yaml | grep -A1 "name: K8S_AWAIT_ELECTION_SERVICE_PORTS_JSON"
kubectl -n d8-linstor get linstorcontrollers.piraeus.linbit.com -o yaml | grep secured
kubectl -n d8-linstor get svc linstor
kubectl -n d8-linstor get endpoints linstor
kubectl -n d8-monitoring get podmonitors.monitoring.coreos.com linstor-controller
kubectl -n d8-monitoring get servicemonitors.monitoring.coreos.com linstor-controller
```
pr4823-kubectl
pr4823-alerts

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AleksZimin. I've started the workflow for label 'e2e/run/static'...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.23 5339009695 5339009695-2-con-1-23 /sys/deckhouse-oss/install:pr4823 astra@192.168.199.98

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.23 5339009695 5339009695-1-con-1-23 /sys/deckhouse-oss/install:pr4823 astra@192.168.199.226

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.23 5339009695 5339009695-1-con-1-23 /sys/deckhouse-oss/install:pr4823 astra@192.168.199.226

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AleksZimin. I've started the workflow for label 'e2e/run/static'...

**Comment by AleksZimin** [comment]
/e2e/abort static;Static;containerd;1.23 5346048743 5346048743-1-con-1-23 /sys/deckhouse-oss/install:pr4823 astra@192.168.199.86

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.23 5346048743 5346048743-1-con-1-23 /sys/deckhouse-oss/install:pr4823 astra@192.168.199.86

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AleksZimin. I've started the workflow for label 'e2e/run/static'...

**Comment by AleksZimin** [comment]
/e2e/abort static;Static;containerd;1.23 5354856155 5354856155-1-con-1-23 /sys/deckhouse-oss/install:pr4823 astra@192.168.199.61

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.23 5354856155 5354856155-1-con-1-23 /sys/deckhouse-oss/install:pr4823 astra@192.168.199.61

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AleksZimin. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AleksZimin. I've started the workflow for label 'e2e/run/static'...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.23 5370593603 5370593603-1-con-1-23 /sys/deckhouse-oss/install:pr4823 astra@192.168.199.147

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.23 5371209884 5371209884-1-con-1-23 /sys/deckhouse-oss/install:pr4823 astra@192.168.199.223

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @duckhawk. I've started the workflow for label 'e2e/run/static'...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.23 5379088002 5379088002-1-con-1-23 /sys/deckhouse-oss/install:pr4823 astra@192.168.199.48

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AleksZimin. I've started the workflow for label 'e2e/run/static'...

**Comment by AleksZimin** [comment]
/e2e/abort static;Static;containerd;1.23 5381058072 5381058072-1-con-1-23 /sys/deckhouse-oss/install:pr4823 astra@192.168.199.85

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.23 5381058072 5381058072-1-con-1-23 /sys/deckhouse-oss/install:pr4823 astra@192.168.199.85

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AleksZimin. I've started the workflow for label 'e2e/run/static'...

**Review by duckhawk — DISMISSED** [review]

**Review by nabokihms — DISMISSED** [review]

File: modules/041-linstor/images/piraeus-operator/patches/040-add-metrics-port.patch
Comment (kvaps) [inline modules/041-linstor/images/piraeus-operator/patches/040-add-metrics-port.patch:1]: Wouldn't it be simpler to specify `K8S_AWAIT_ELECTION_SERVICE_PORTS_JSON` in `spec.additionalEnv` for linstor-controller resource?

Comment (AleksZimin) [reply modules/041-linstor/images/piraeus-operator/patches/040-add-metrics-port.patch:1]: I have tried to specify K8S_AWAIT_ELECTION_SERVICE_PORTS_JSON in spec.additionalEnv but it does not override the value that is hardcoded in the operator.

**Review by kvaps — APPROVED** [review]
LGTM

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [5083](https://github.com/deckhouse/deckhouse/pull/5083) to the branch [release-1.47](https://github.com/deckhouse/deckhouse/tree/release-1.47) successful!

**Pull request merged**